app/assets/javascripts/short-form/ShortFormNavigationService.js.coffee

```diff
@@ -272,6 +275,8 @@ ShortFormNavigationService = (
           'preferences-programs'
       when 'review-submitted'
         'confirmation'
+      when 'name'
```

Comment:
why don't we have to put the check for isSale here?

Reply:
Because on name page on rental listing, back button is not displayed.